diff --git a/plugins/content_types/menu_tree/menu_tree.inc b/plugins/content_types/menu_tree/menu_tree.inc
index e258fe3..bf2aed1 100644
--- a/plugins/content_types/menu_tree/menu_tree.inc
+++ b/plugins/content_types/menu_tree/menu_tree.inc
@@ -133,6 +133,9 @@ function menu_block_menu_tree_content_type_edit_form_submit(&$form, &$form_state
       $form_state['conf'][$key] = $form_state['values'][$key];
     }
   }
+  if (!empty($form_state['values']['parent'])) {
+    list($form_state['conf']['menu_name'], $form_state['conf']['parent_mlid']) = explode(':', $form_state['values']['parent']);
+  }  
 }
 
 /**
